src/features/ListAllMovies/ui/MoviesList.tsx

```diff
@@ -35,21 +46,35 @@ export const MoviesList: FC<MovieListProp> = ({ type, limit }) => {
         return <div>Error loading movies. Please try again later.</div>;
     }
 
+    const items = data?.pages.flatMap((page) => splitToRows(page.items, 6));
+
     return (
         <>
             <MovieFilters data={data} setFilters={setFilters} />
-            <div className="flex gap-8 flex-wrap mt-5 mb-8 pl-10 max-w-[1400px]">
-                {data?.pages.map((page, idx) => (
+            <div className="flex gap-8 flex-wrap mt-5 pl-10 max-w-[1400px]">
+                {items?.map((page, idx) => (
                     <Fragment key={idx}>
-                        {page.items.map((movie) => (
+                        {page.map((movie) => (
                             <MovieItem
                                 key={movie.id}
                                 genre={movie.genre}
                                 duration={movie.duration}
                                 title={movie.title}
                                 imageUrl={movie.imageUrl}
+                                description={movie.description}
+                                rowIndex={idx}
+                                setSelectedMovie={setSelectedMovie}
```

Comment:
It's a good practise to concentrate your logic in one place if you have such ability
For example here we have MoviesList which contains MovieItem and MovieDetails
Right now you are adding a new functionality for selecting particular movie for showing the whole info 
Ideally MovieItem shouldn't know what exactly will happened by clicking on it, so it's better to
1. instead of using `setSelectedMovie` prop use more generic prop `onClick` and use it inside
2. there is no sense to pass rowIndex to MovieItem, because MovieItem can be used out of array and actually you just put the index to the click callback, so the index is not used inside MovieItem to run some logic. You can just put the index here like this: onClick={() => { setSelectedMovie({ ...movie, rowIndex: idx }) }} or you can even create a separate method here inside MoviesList and pass to onClick prop

So the idea is to concentrate all functionality within MoviesList and pass needed callbacks to components